pkg/datapath/iptables/iptables.go

```diff
@@ -101,6 +101,29 @@ func getFeedRule(name, args string) []string {
 	return append(argsList, ruleTail...)
 }
 
+// KernelHasNetfilter probes whether iptables related modules are present in
+// the kernel and returns true if indeed the case, else false.
+func KernelHasNetfilter() bool {
+	modulesManager := &modules.ModulesManager{}
+	ip6tables := true
+	iptables := true
+
+	if err := modulesManager.Init(); err != nil {
+		return true
+	}
+
+	if found, _ := modulesManager.FindModules(
+		"ip_tables", "iptable_mangle", "iptable_raw", "iptable_filter"); !found {
```

Comment:
Couldn't this be `if found { return true }`? same for the v6 case.

Also, shouldn't we execute this line based on the `option.Config.IPv4Enabled` and same for the `V6`?

Reply:
Left latter as-is because even if v4 modules are present but no v6 ones and I'd only have v6 enabled, then we could still be subject to ct invalid state drops.